File: src/test/java/quebec/virtualite/backend/services/rest/CityDTOTest.java
package quebec.virtualite.backend.services.rest;

import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

import static quebec.virtualite.backend.TestConstants.BAD_CITY_DTO;
import static quebec.virtualite.backend.TestConstants.CITY_DTO;
import static quebec.virtualite.backend.services.utils.TestUtils.assertInvalid;
import static quebec.virtualite.backend.services.utils.TestUtils.assertValid;

@ExtendWith(MockitoExtension.class)
@Tag("UnitTest")
class CityDTOTest
{
    @Test
    void validate()
    {
        assertValid(CITY_DTO);

        assertInvalid(CITY_DTO.withProvince(null));
        assertInvalid(CITY_DTO.withProvince(""));

        assertInvalid(CITY_DTO.withName(null));
        assertInvalid(CITY_DTO.withName(""));

        assertInvalid(BAD_CITY_DTO);
    }
}

File: src/test/java/quebec/virtualite/backend/services/rest/CityDTOTest.kt
package quebec.virtualite.backend.services.rest

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.junit.jupiter.MockitoExtension
import quebec.virtualite.backend.TestConstants.BAD_CITY_DTO
import quebec.virtualite.backend.TestConstants.CITY_DTO
import quebec.virtualite.backend.services.utils.TestUtils.assertInvalid
import quebec.virtualite.backend.services.utils.TestUtils.assertValid

@ExtendWith(MockitoExtension::class)
class CityDTOTest
{
    @Test
    fun validate()
    {
        assertValid(CITY_DTO)

        assertInvalid(CITY_DTO.copy(name = null))
        assertInvalid(CITY_DTO.copy(name = ""))

        assertInvalid(CITY_DTO.copy(province = null))
        assertInvalid(CITY_DTO.copy(province = ""))

        assertInvalid(BAD_CITY_DTO)
    }
}
